parser/errors.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parser
+
+import (
+	"errors"
+
+	utils "github.com/coinbase/rosetta-sdk-go/errors"
+)
+
+// Named error types for Parser errors
+var (
+	///////////////////
+	/* INTENT ERRORS */
+	///////////////////
+	ErrExpectedOperationAccountMismatch = errors.New(
+		"intended account did not match observed account",
+	)
+	ErrExpectedOperationAmountMismatch = errors.New(
+		"intended amount did not match observed amount",
+	)
+	ErrExpectedOperationTypeMismatch = errors.New(
+		"intended type did not match observed type",
+	)
+	ErrExpectedOperationsUnexpectedOperation = errors.New("found extra operation")
+	ErrExpectedSignerDuplicateSigner         = errors.New("found duplicate signer")
+	ErrExpectedSignerUnexpectedSigner        = errors.New("found unexpected signers")
+
+	IntentErrs = []error{
+		ErrExpectedOperationAccountMismatch,
+		ErrExpectedOperationAmountMismatch,
+		ErrExpectedOperationTypeMismatch,
+		ErrExpectedOperationsUnexpectedOperation,
+		ErrExpectedSignerDuplicateSigner,
+		ErrExpectedSignerUnexpectedSigner,
+	}
+
+	/////////////////////////////
+	/* MATCH OPERATIONS ERRORS */
+	/////////////////////////////
+	ErrAccountMatchAccountMissing           = errors.New("account is missing")
+	ErrAccountMatchSubAccountMissing        = errors.New("SubAccountIdentifier.Address is missing")
+	ErrAccountMatchSubAccountPopulated      = errors.New("SubAccount is populated")
+	ErrAccountMatchUnexpectedSubAccountAddr = errors.New("unexpected SubAccountIdentifier.Address")
+
+	ErrMetadataMatchKeyNotFound       = errors.New("key is not present in metadata")
+	ErrMetadataMatchValueTypeMismatch = errors.New("unexpected value associated with key")
```

Comment:
nit: the name and the error message doesn't seem to match 🤔

Reply:
changed to `ErrMetadataMatchKeyValueMismatch`